airbyte-integrations/connectors/source-netsuite/source_netsuite/streams.py

```diff
@@ -0,0 +1,209 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from collections import OrderedDict
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+# Basic full refresh stream
+import requests
+import zeep
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+from netsuitesdk import NetSuiteConnection
+
+
+class NetsuiteSoapStream(HttpStream):
+
+    def __init__(self, nc: NetSuiteConnection, config: dict, concurrency_limit: int = 1):
+        self.config = config
+        super().__init__()
+
+        self.nc = nc
+
+        self.concurrency_limit = concurrency_limit
+
+        self.entities_will_have_array_structure = ["VendorBill", "VendorPayment","JournalEntry", "ExpenseReport", "ExpenseCategories"]
+
+    url_base = "/"
+
+    @staticmethod
+    def as_serialized(data) -> Iterable[Mapping[str, Any]]:
+        return [zeep.helpers.serialize_object(_) for _ in data]
+
+    def is_entities_or_dicts_inside_list(self, data: list) -> bool:
+        first_element = data[0]
+        if isinstance(first_element, OrderedDict):
+            return False
+        elif isinstance(first_element, list):
+            self.is_entities_or_dicts_inside_list(first_element)
+        return True
```

Comment:
This could be simplified as follows:
```python
def is_entities_or_dicts_inside_list(self, data: list) -> bool:
        # getting first element, because of <explain> ?
        # depending of the structure of returned data, we <explain> ?
        if isinstance(data[0], OrderedDict):
            return False
        elif isinstance(data[0], list):
            self.is_entities_or_dicts_inside_list(data[0])
        return True
```